woocommerce/woocommerce-ios
# [UL&S] Login with site address: Jetpack not installed or not connected

Closes #3130 
Closes #3131 

Ref: p91TBi-3ON-p2

⚠️ Please note this PR is not against develop, but against the feature branch for UL&S 🙇

##Design
<img src="https://user-images.githubusercontent.com/2722505/100309701-75c8a780-2fe6-11eb-92da-bb89814db8f0.png" width="350"/>

## Changes

In action:
<img src="https://user-images.githubusercontent.com/2722505/100309777-97299380-2fe6-11eb-9537-b639b6dc9983.gif" width="350"/>

Screenshots:

| | Light Appearance | Dark Appearance |
| ---- | ---- | ---- |
| iPhone SE | <img src="https://user-images.githubusercontent.com/2722505/100309876-db1c9880-2fe6-11eb-9280-3ad7e0599c05.png" width="350"/> | <img src="https://user-images.githubusercontent.com/2722505/100309878-dd7ef280-2fe6-11eb-8b93-791bbe2d482b.png" width="350"/> |
| iPhone 12 | <img src="https://user-images.githubusercontent.com/2722505/100309912-f1c2ef80-2fe6-11eb-8fa1-ce5818364697.png" width="350"/> | <img src="https://user-images.githubusercontent.com/2722505/100309907-eec7ff00-2fe6-11eb-96e8-3e9a5c6dc37b.png" width="350"/> |
| iPhone 12 Pro Max | <img src="https://user-images.githubusercontent.com/2722505/100309952-02736580-2fe7-11eb-8be8-bbdbb37e72f2.png" width="350"/> | <img src="https://user-images.githubusercontent.com/2722505/100309946-00110b80-2fe7-11eb-9bea-cc3f88980816.png" width="350"/> |

⚠️  Adding the help button requires another update to WPAuthenticator, so I have deferred it to a separate issue: #3214

## How it works
The heavy lifting is done by WordPressAuthenticator. What we do is provide a View Controller to the library, and let it inset it in whatever way it makes sense. See p91TBi-3ON-p2 for more context.

This PR:
* Points the Podfile to a specific beta release of WPAuthenticator.
* Adds a new view controller (ULErrorViewController). All error messages for UL&S seem to be modeled using the same UI pattern, so I decided to go "generic", and attempt to implement all these special screens (that contain an image, text, a text button and two larger buttons) using the same view controller
* The way to customize this ULErrorViewController is by injecting a view model that provides the content (i.e. text, button titles) and that handles button actions.
* Adds unit tests

## Testing
* Checkout the branch. Run `bundle exec pod install`
* Log out if necessary
* Attempt to Log In with Store Address. Enter a store address of a site that does not have Jetpack installed or does not have Jetpack connected.
* Notice if the error screen in shown. 
* The "What is Jetpack" button should load a web view pointing to this article: https://jetpack.com/about/ * The "See instructions" button should load a web view pointing to: https://docs.woocommerce.com/document/jetpack-setup-instructions-for-the-woocommerce-mobile-app/
* The "Refresh after install" button should navigate to the screen where users enter the site address. It is not actually possible to refresh this screen, so there isn't much more we can do, as far as I can tell. Better solutions are welcome!

Update release notes:
